src/frontend/src/app/pages/proposal-list/proposal-list.component.ts

```diff
@@ -0,0 +1,342 @@
+import { Dialog } from '@angular/cdk/dialog';
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { RouterLink } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+
+import { CardComponent } from '@cg/angular-ui';
+import { ReviewPeriodState } from '@cg/backend';
+import { FormatDatePipe } from '~core/pipes';
+import {
+  ProposalLinkBaseUrl,
+  ProposalService,
+  ProposalTopic,
+} from '~core/state';
+import {
+  KeyValueGridComponent,
+  KeyColComponent,
+  ValueColComponent,
+  FormFieldComponent,
+  InputDirective,
+  LabelDirective,
+  LoadingDialogComponent,
+  getLoadingDialogConfig,
+  LoadingDialogInput,
+} from '~core/ui';
+import { keysOf } from '~core/utils';
+
+enum FilterOptions {
+  InReview = 'In review',
+  Reviewed = 'Reviewed',
+  All = 'All',
+}
+
+@Component({
+  selector: 'app-proposal-list',
+  standalone: true,
+  imports: [
+    CommonModule,
+    ReactiveFormsModule,
+    KeyValueGridComponent,
+    KeyColComponent,
+    ValueColComponent,
+    FormFieldComponent,
+    InputDirective,
+    LabelDirective,
+    FormatDatePipe,
+    RouterLink,
+    CardComponent,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styles: [
+    `
+      @import '@cg/styles/common';
+
+      :host {
+        @include page-content;
+      }
+
+      .proposal {
+        margin-bottom: size(6);
+      }
+
+      .proposal__title,
+      .proposal__proposer {
+        word-break: break-word;
+      }
+
+      .proposal__link {
+        margin-right: size(4);
+      }
+
+      .proposal__vote {
+        font-weight: bold;
+      }
+
+      .proposal__vote--adopt {
+        color: $success;
+      }
+
+      .proposal__vote--reject {
+        color: $error;
+      }
+
+      .filter {
+        display: flex;
+        flex-direction: column;
+        margin-bottom: size(4);
+
+        @include sm {
+          flex-direction: row;
+        }
+      }
+
+      .filter__label {
+        margin-bottom: size(2);
+        margin-right: size(2);
+        width: 100%;
+
+        @include sm {
+          width: 50%;
+        }
+      }
+    `,
+  ],
+  template: `
+    <h1 class="h3">Proposals</h1>
+    <form [formGroup]="filterForm">
+      <div class="filter">
+        <div class="filter__label">Select code review status:</div>
+        <app-form-field class="filter__value">
+          <div class="radio-group">
+            @for (key of filterOptionsKeys; track key) {
+              <input
+                appInput
+                [id]="key"
+                [value]="key"
+                type="radio"
+                formControlName="filter"
+                (change)="updateFilter()"
+              />
+              <label appLabel [for]="key">
+                {{ filterOptions[key].valueOf() }}
```

Comment:
I'm not really a fan of treating the enum like an object, I'd prefer to just create an object for this. I think it'll be easier for us to discuss offline.